portal_tables/sync_education.py

```diff
@@ -5,12 +5,48 @@
 """
 
 import pandas as pd
+import re
 import utils
 
+def add_missing_info(
+    education: pd.DataFrame) -> pd.DataFrame:
+    """Add missing information into table before syncing."""
+
+    url_pattern = re.compile(".*(synapse\.org).*")
+    alias_pattern = re.compile("^syn\d{7,8}$")
+    education["synapseLink"] = ""
+    for _, row in education.iterrows():
+        is_in_synapse = None
+        alias_list = row["ResourceAlias"].split(",")
+        link_list = row["ResourceLink"].split(",")
+        syn_link_list = []
+        for a in alias_list:
+            a_match = re.match(alias_pattern, a)
+            is_in_synapse = True if a_match else None
+            if is_in_synapse:
+                syn_link = "".join(["https://www.synapse.org/Synapse:", a])
+                formatted_syn_link = "".join(["[", a, "](", syn_link, ")"])
+                syn_link_list.append(formatted_syn_link)
+                syn_links = ",".join(syn_link_list)
+                education.at[_, "synapseLink"] = syn_links
+            else:
+                for s in link_list:
+                    s_match = re.match(url_pattern, s)
+                    is_in_synapse = True if s_match else None
+                    if is_in_synapse:
+                        education.at[_, "synapseLink"] = "".join(["[Link](", education.at[_, "ResourceLink"], ")"])
```

Comment:
At a quick glance, there is a lot going on here 😅  I think using list comprehensions and f-strings could help improve the readability.